main.go

```diff
@@ -66,6 +69,15 @@ func main() {
 		exit(1)
 	}
 
+	hasher := sha256.New()
+	s, err := ioutil.ReadFile(path)
```

Comment:
I think it's not so nice to open the same file multiple times. How about something like this:
```go
hasher := sha256.New()
_, err = io.Copy(hasher, f)
if err != nil {
	fmt.Fprintf(stderr, "Error reading '%s': %s\n", "path", err.Error())
	exit(1)
}
f.Seek(0, 0)
```

Reply:
I agree, it's been fixed